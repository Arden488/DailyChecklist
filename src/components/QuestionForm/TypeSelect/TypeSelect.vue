<template>
  <el-select v-model="typeLocal" @change="updateType($event)" placeholder="Select one">
    <el-option label="Time picker" value="time"></el-option>
    <el-option label="Yes/No" value="boolean"></el-option>
    <el-option label="Select variant" value="select"></el-option>
    <el-option label="Mood rating" value="mood"></el-option>
  </el-select>
</template>

<script>
export default {
  name: 'TypeSelect',
  props: [ 'type' ],
  data () {
    return {
      typeLocal: this.type,
    }
  },
  methods: {
    updateType: function (value) {
      // console.log(value);
      this.$emit('type', value);
    }
  }
}
</script>
