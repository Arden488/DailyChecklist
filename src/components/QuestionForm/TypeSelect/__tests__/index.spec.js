/* global it, describe, expect */

import { shallowMount, mount, createLocalVue } from '@vue/test-utils';
import Vue from 'vue';
import VueRouter from 'vue-router';
import ElementUI from 'element-ui';
import TypeSelect from '../TypeSelect.vue';

const localVue = createLocalVue();
localVue.use(ElementUI);
localVue.use(VueRouter);

describe('Default layout', () => {
  it('is a Vue instance', () => {
    const wrapper = shallowMount(TypeSelect, { localVue });
    expect(wrapper.isVueInstance).toBeTruthy()
  })

  it('is renders correctly', () => {
    const wrapper = shallowMount(TypeSelect, { localVue });
    expect(wrapper).toMatchSnapshot()
  })

  it('it\'s type is empty', () => {
    const wrapper = shallowMount(TypeSelect, { localVue });
    console.log(wrapper.vm.props);
    expect(wrapper.vm.typeLocal).toBe('');
  })
});
