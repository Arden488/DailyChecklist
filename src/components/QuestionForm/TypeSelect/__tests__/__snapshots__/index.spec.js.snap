// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Default layout is renders correctly 1`] = `
<div class="el-select">
  <!---->
  <div class="el-input el-input--suffix">
    <!---->
    <input type="text" readonly="readonly" autocomplete="off" placeholder="Select one" class="el-input__inner">
    <!----><span class="el-input__suffix"><span class="el-input__suffix-inner"><i class="el-select__caret el-input__icon el-icon-arrow-up"></i><!----></span>
    <!---->
    </span>
    <!---->
  </div>
  <div class="el-select-dropdown el-popper" style="display: none;">
    <div class="el-scrollbar" style="">
      <div class="el-select-dropdown__wrap el-scrollbar__wrap el-scrollbar__wrap--hidden-default">
        <ul class="el-scrollbar__view el-select-dropdown__list">
          <!---->
          <li class="el-select-dropdown__item"><span>Time picker</span></li>
          <li class="el-select-dropdown__item"><span>Yes/No</span></li>
          <li class="el-select-dropdown__item"><span>Select variant</span></li>
          <li class="el-select-dropdown__item"><span>Mood rating</span></li>
        </ul>
      </div>
      <div class="el-scrollbar__bar is-horizontal">
        <div class="el-scrollbar__thumb" style="transform: translateX(0%); -webkit-transform: translateX(0%);"></div>
      </div>
      <div class="el-scrollbar__bar is-vertical">
        <div class="el-scrollbar__thumb" style="transform: translateY(0%); -webkit-transform: translateY(0%);"></div>
      </div>
    </div>
    <!---->
  </div>
</div>
`;

exports[`TypeSelect component is renders correctly 1`] = `
<div class="el-select">
  <!---->
  <div class="el-input el-input--suffix">
    <!---->
    <input type="text" readonly="readonly" autocomplete="off" placeholder="Select one" class="el-input__inner">
    <!----><span class="el-input__suffix"><span class="el-input__suffix-inner"><i class="el-select__caret el-input__icon el-icon-arrow-up"></i><!----></span>
    <!---->
    </span>
    <!---->
  </div>
  <div class="el-select-dropdown el-popper" style="display: none;">
    <div class="el-scrollbar" style="">
      <div class="el-select-dropdown__wrap el-scrollbar__wrap el-scrollbar__wrap--hidden-default">
        <ul class="el-scrollbar__view el-select-dropdown__list">
          <!---->
          <li class="el-select-dropdown__item"><span>Time picker</span></li>
          <li class="el-select-dropdown__item"><span>Yes/No</span></li>
          <li class="el-select-dropdown__item"><span>Select variant</span></li>
          <li class="el-select-dropdown__item"><span>Mood rating</span></li>
        </ul>
      </div>
      <div class="el-scrollbar__bar is-horizontal">
        <div class="el-scrollbar__thumb" style="transform: translateX(0%); -webkit-transform: translateX(0%);"></div>
      </div>
      <div class="el-scrollbar__bar is-vertical">
        <div class="el-scrollbar__thumb" style="transform: translateY(0%); -webkit-transform: translateY(0%);"></div>
      </div>
    </div>
    <!---->
  </div>
</div>
`;
